Automattic/jetpack
# CRM: Accept Quote button problem for php versions 8.1 and up

For PHP versions 8.1 and up, the accept quote button is not working as expected. It always shows an error message, even when it succeeds in accepting the quote (a page refresh after an error message reveals that the quote was accepted).

This PR fixes this problem.

## Proposed changes:

* The problem was happening because of a deprecation notice when calling the constructor of the class `DateTime` passing null as the first argument. As per the official documentation, to get the current datetime the string `'now'` has to be passed as first argument.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Create a contact (e.g. `wp-admin/admin.php?page=zbs-add-edit&action=edit&zbstype=contact`)
* Add this contact to the Client Portal by giving access to them using the Client Portal panel in `wp-admin/admin.php?page=zbs-add-edit&action=view&zbstype=contact&zbsid=1`
* Add some quotes (and make sure they are not accepted) to this contact
* Log in as the contact
* Go to the Client Portal (by default `/clients`)
* Go to the quote listing (by default `/clients/quotes`)
* Select a quote and try to accept it

In `trunk` you will get an error message, and when trying to refresh the page the quote will be accepted.
In `fix/crm-accept-quote-button-problem-php-81-and-82` the message will be the success message (` Quote accepted, Thank you. `).
